### PR TITLE
Added new traceability fields to deployment report

### PR DESCRIPTION
**Pull request by Josh-01**

### CODE REVIEW

**Review by Ganeshrockz — APPROVED** [review]

File: src/utilities/utility.ts [hunk 1]
```diff
@@ -129,6 +138,54 @@ export function annotateChildPods(kubectl: Kubectl, resourceType: string, resour
...
+            pathValue = pathLink;
+        }
+        return pathValue;
+    });
+
+    let inputManifestFiles: string[] = [];
+    if (!helmChartPaths || helmChartPaths.length == 0) {
+        let inputManifestFiles = inputParams.manifests || [];
+        inputManifestFiles.map(manifestFile => {
+            let pathValue = manifestFile;
```
Comment (ajinkya599) [inline src/utilities/utility.ts:157]: Move this logic to a utility method and reuse for all the cases (Helm, Manifest, Docker).
Something like - `getNormalizedPath(path)` that checks if it is already a URL or not, and returns the URL.

File: src/utilities/utility.ts [hunk 2]
```diff
@@ -129,6 +138,54 @@ export function annotateChildPods(kubectl: Kubectl, resourceType: string, resour
...
+
+    const deploymentConfig = <DeploymentConfig>{
+        manifestFilePaths: inputManifestFiles,
+        helmChartFilePaths: helmChartPaths,
+        dockerfilePaths: imageDockerfilePathMap
+    };
+
+    return Promise.resolve(deploymentConfig);
+}
+
```
Comment (ajinkya599) [inline src/utilities/utility.ts:187]: Remove extra space

File: src/utilities/utility.ts [hunk 3]
```diff
@@ -129,6 +138,54 @@ export function annotateChildPods(kubectl: Kubectl, resourceType: string, resour
     return commandExecutionResults;
 }
 
+export async function getDeploymentConfig(): Promise<DeploymentConfig> {
+
+    let helmChartPaths = (process.env.HELM_CHART_PATHS && process.env.HELM_CHART_PATHS.split('\n').filter(path => path != "")) || [];
```
Comment (ajinkya599) [inline src/utilities/utility.ts:143]: @ds-ms suggested that we use `;` as a delimiter. `\n` doesn't work while setting variables.
Good to `trim` the extracted path. (Using `map` after filter?)

File: src/utilities/utility.ts (hunk 1 above)
Comment (Josh-01) [reply src/utilities/utility.ts:157]: Done

File: src/utilities/utility.ts (hunk 3 above)
Comment (Josh-01) [reply src/utilities/utility.ts]: Done

File: src/utilities/utility.ts (hunk 2 above)
Comment (Josh-01) [reply src/utilities/utility.ts:176]: Done

File: src/traceability/azure-traceability-helper.ts [hunk 4]
```diff
@@ -99,7 +100,9 @@ function getResourceUri(aksResourceContext: AksResourceContext): string {
   return `${aksResourceContext.managementUrl}subscriptions/${aksResourceContext.subscriptionId}/resourceGroups/${aksResourceContext.resourceGroup}/providers/Microsoft.Devops/deploymentv2/${deploymentName}?api-version=2020-10-01-preview`;
 }
 
-function getDeploymentReport(context: AksResourceContext, kubectl: Kubectl) {
```
Comment (ajinkya599) [inline src/traceability/azure-traceability-helper.ts:102]: Seems like some merge issue..Include `kubectl` parameter.

File: src/traceability/azure-traceability-helper.ts [hunk 5]
```diff
@@ -79,8 +80,8 @@ async function createDeploymentResource(aksResourceContext: AksResourceContext,
 
 export async function addTraceability(kubectl: Kubectl): Promise<void> {
   const aksResourceContext = getAksResourceContext();
-  if (aksResourceContext !== null) {
```
Comment (ajinkya599) [inline src/traceability/azure-traceability-helper.ts:82]: Incorporate the new changes.

**Review by ajinkya599 — APPROVED** [review]
🚀

**Pull request merged**